blacktie/tx/src/main/cpp/HttpTxManager.cxx

```diff
@@ -379,7 +376,7 @@ bool HttpTxManager::handle_request(
 					LOG4CXX_DEBUG(httptxlogger, "Return 200 OK with body: " <<
 						status << " XA status: " << res);
 					code = 200;
-					_branches.erase(branch->get_name());
+					_branches.erase(wid);
```

Comment:
So do you agree that the fix is wrong. I still think we need to find the root cause. The intent is to clean up branches from two places:

* during normal processing XAWrapper::set_complete does the clean up
* during recovery XAResourceManager::recover does the clean up

What do you mean by your second comment. Can you point me at the exact bit of code you are referring to.

XAResourceManager::recoverRTS